satellite/satellitedb/dbx/satellitedb.dbx

```diff
@@ -680,3 +680,37 @@ update offer (
 )
 
 create offer ( )
+
+
+//--- user credit table ---//
+
+model user_credit (
+	key	id
+
+	field id				serial
+
+	field user_id user.id	restrict
+	field offer_id offer.id	restrict
+	field referred_by user.id	restrict ( nullable )
+
+	field credits_earned_in_cents	int
+	field credits_used_in_cents	int ( updatable, autoinsert )
+
+	field expires_at		timestamp	( updatable )
+	field created_at		timestamp	( autoinsert )
+)
+
+read all (
```

Comment:
maybe it would be better to write your own query for this in something that wraps the dbx stuff? The functions that dbx autogenerates from this look pretty crazy/verbose.
Here is an example of what I am saying you might be able to do: https://github.com/storj/storj/blob/master/satellite/satellitedb/overlaycache.go#L360-L380